Tests/UnitTests/OrderTests.swift

```diff
@@ -0,0 +1,89 @@
+//
+// OrderTests.swift
+//
+// Copyright © 2019 Button, Inc. All rights reserved. (https://usebutton.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+	
+import XCTest
+@testable import ButtonMerchant
+
+class OrderTests: XCTestCase {
```

Comment:
Lets see some test passing nil variables to the requred arguments

Reply:
There's no support for defaults in Obj-C so they need to explicitly pass in all values from the init method.

Reply:
I assume this is in reference to my other question?

There are quite a lot more arguments to the init now, which will make the objective c api considerably worse. Let’s remove the optional arguments.

Reply:
If we remove the arguments for the optional properties, we would need to change the properties from `let` to `var` so that they can be set with dot notation. Is this what you're expecting?